fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/api/client/rendereregistry/v1/item/CooldownOverlayProperties.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendereregistry.v1.item;
+
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.impl.client.renderer.registry.DefaultCooldownOverlayProperties;
+
+public interface CooldownOverlayProperties {
+	CooldownOverlayProperties DEFAULT = new DefaultCooldownOverlayProperties();
+
+	boolean isVisible(ItemStack stack);
+	float getFillFactor(ItemStack stack);
```

Comment:
maybe `getProgress`/`getFillProgress`?

Reply:
Since this is purely graphical and the overlay can be used to represent anything (not just cooldowns), I think fill factor is fine. It should just say how much of the item is filled with the overlay.